hangman.py

```diff
@@ -0,0 +1,311 @@
+    avaible_letters.sort()
+    return str(avaible_letters).replace(", ", "").replace("'","")[1:-1]
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+
+
+
+
+def hangman(secret_word):
+    '''
+    secret_word: string, the secret word to guess.
+    
+    Starts up an interactive game of Hangman.
+    
+    * At the start of the game, let the user know how many 
+      letters the secret_word contains and how many guesses s/he starts with.
+      
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+    
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+    
+    * The user should receive feedback immediately after each guess 
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the 
+      partially guessed word so far.
+    
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    warnings = 3
+    attemps = 6
+    letters_guessed = set()
+    while True:
+        print("_" * 16)
+        print("Warnings: ", warnings)
+        print("Available letters: ",get_available_letters(letters_guessed))
+        print("Attemps: ", attemps)
+        letter = input("enter letter:").strip(' ')
+        letter.lower()
+        if not letter.isalpha() or letter in letters_guessed:
+            if warnings >0:
+                warnings -=1
+                print("Oooops! It`s a danger!!!")
+            else:
+                attemps -=1
+        else:
+            letters_guessed.add(letter)
+            splited_word = get_guessed_word(secret_word, letters_guessed)
+            print(splited_word)
+            guessed = splited_word
+            if is_word_guessed(secret_word,letters_guessed):
+                print("You win!!!")
+                break
+            if letter not in secret_word:
+                if letter in "aeoiu":
+                    attemps -= 1
+                attemps -= 1
+        if attemps <= 0:
+            print("You lose!")
+            break
+
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+
+
+
+# When you've completed your hangman function, scroll down to the bottom
+# of the file and uncomment the first two lines to test
+#(hint: you might want to pick your own
+# secret_word while you're doing your own testing)
+
+
+# -----------------------------------
+
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+    my_word_1 = my_word.replace(" ","")
```

Comment:
It worth to have this replacement on 1 level up (as `my_word` is almost always the same)